config/common.yml

```diff
@@ -0,0 +1,22 @@
+services:
+  demodeprecated.module:
+    class: 'DemoDeprecated'
+    factory: ['Module', 'getInstanceByName']
```

Comment:
Ooooh nice trick

it could go in the FAQ of the devdocs 😄 

https://devdocs.prestashop.com/1.7/faq/